### PR TITLE
feat(masthead-l1): Removing the has pseudo-class

### PR DESCRIPTION
**Pull request by pjudge**
### Related Ticket(s)

{{Provide url(s) to the related ticket(s) that this pull request addresses}}

### Description

The `:has` pseudo-class is not supported across all browsers yet (lookin' at you Firefox). This PR removes the usage of `:has` inside the masthead L1 styles, without changing the look on the front end. 

### Changelog

**Changed**

- Added a `classMap` to the L1 dropdown links element to dynamically add a class if the dropdown contains a wide column. 
- Added a similar functionality to the pre-existing headingClasses `classMap` to account for when headings inside dropdowns are not links. 
- Updated the corresponding styles to use the new classes instead of `:has`.

### CODE REVIEW

**Review by kennylam — COMMENTED** [review]
@pjudge Thanks for this! The only thing I'd suggest is to use BEM naming conventions, e.g. something like

```css
.#{$prefix}--masthead__l1-dropdown-item--no-link
.#{$prefix}--masthead__l1-dropdown--column-wide
```

File: packages/web-components/src/components/masthead/masthead-l1.ts [hunk 1]
```diff
@@ -752,8 +756,15 @@ class DDSMastheadL1 extends StableSelectorMixin(LitElement) {
...
 
+    let noLink: boolean = false;
+
+    if (!heading.url) {
+      noLink = true;
+    }
+
     const headingClasses = classMap({
       [`${prefix}--masthead__l1-dropdown-heading`]: true,
+      [`no-link`]: noLink,
```
Comment (andy-blum) [inline packages/web-components/src/components/masthead/masthead-l1.ts:767]: As @kennylam mentioned, let's BEM-ify this class name.

For the sake of learning here, though:

Readability boost: we could reduce this all to a single line here.

```
const headingClasses = classMap({
      [`${prefix}--masthead__l1-dropdown-heading`]: true,
      "no-link": Boolean(heading.url) === false,
});
```
The `[\`...\`]` version is only necessary if we want to do the string interpolation like in the line above with `${prefix}` (which we will when we make a BEM version)

We can also do simple JS comparisons on the value side of the key/value pair, making the `noLink` variable and it's logic above unnecessary.

File: packages/web-components/src/components/masthead/masthead-l1.ts [hunk 2]
```diff
@@ -400,6 +400,10 @@ class DDSMastheadL1 extends StableSelectorMixin(LitElement) {
 
     const wideColumns = menuSections.filter((section) => section.span > 1);
     const normalColumns = menuSections.filter((section) => !(section.span > 1));
+    const dropdownClasses = classMap({
+      [`${prefix}--masthead__l1-dropdown-links`]: true,
+      [`has-wide`]: hasWideColumn,
```
Comment (andy-blum) [inline packages/web-components/src/components/masthead/masthead-l1.ts:405]: See comments at line 767

**Review by andy-blum — CHANGES_REQUESTED** [review]

**Comment by pjudge** [comment]
Thank you @kennylam and @andy-blum ! I have made the updates you suggested. 👍

**Review by kennylam — APPROVED** [review]
LGTM!

**Review by andy-blum — APPROVED** [review]

**Comment by andy-blum** [comment]
e2e test failure seems unrelated. Re-running.
Screenshot 2023-06-01 at 4 12 38 PM

**Pull request merged**